service/src/event/service.rs

```diff
@@ -197,68 +197,72 @@ impl CeramicEventService {
         let ordered = OrderEvents::try_new(&self.pool, to_insert).await?;
 
         let missing_history = ordered
-            .missing_history
+            .missing_history()
             .iter()
             .map(|(e, _)| e.order_key.clone())
             .collect();
 
-        let to_insert_with_metadata = if history_required {
-            ordered.deliverable
+        // api writes shouldn't have any missed history so we don't insert those events and
+        // we can skip notifying the ordering task because it's impossible to be waiting on them
+        let store_result = if history_required {
+            let to_insert = ordered.deliverable().iter().map(|(e, _)| e);
+            CeramicOneEvent::insert_many(&self.pool, to_insert).await?
         } else {
-            ordered
-                .deliverable
-                .into_iter()
-                .chain(ordered.missing_history)
-                .collect()
-        };
+            let to_insert = ordered
+                .deliverable()
+                .iter()
+                .map(|(e, _)| e)
+                .chain(ordered.missing_history().iter().map(|(e, _)| e));
 
-        let to_insert = to_insert_with_metadata
-            .iter()
-            .map(|(e, _)| e.clone())
-            .collect::<Vec<_>>();
+            let store_result = CeramicOneEvent::insert_many(&self.pool, to_insert).await?;
+            self.notify_ordering_task(&ordered, &store_result).await?;
 
-        let res = CeramicOneEvent::insert_many(&self.pool, &to_insert[..]).await?;
+            store_result
+        };
 
-        // api writes shouldn't have any missed pieces that need ordering so we don't send those and return early
-        if history_required {
-            return Ok(InsertResult {
-                store_result: res,
-                missing_history,
-            });
-        }
+        Ok(InsertResult {
+            store_result,
+            missing_history,
+        })
+    }
```

Comment:
I really like how this logic has simplified. It very clear how the two branches differ in behavior with and without history being required.